### DISCUSSION TITLE
What the parameter's definition of the function which is the first parameter of `beforeAll`?

### DISCUSSION THREAD

**Discussion by yuzheng14** [vitest-dev/vitest#6067] [Q&A] [answered]
The `playground/vitestSetup.ts` file in Vite's repo use `beforeAll` like following picture:

![image](https://github.com/vitest-dev/vitest/assets/55919198/c333910e-e4a4-4cb7-ba86-2c5ceb0a0c45)

But I couldn't find any explanation of this `s` parameter. And it actually in the type definition of `@vite/runner`

https://github.com/vitest-dev/vitest/blob/7012f8c12a3d0004e7e7651bbeee905ac9102098/packages/runner/src/types/tasks.ts#L369-L379

**Comment by sheremet-va** [comment]
It's `File` or `Suite` from `vitest`:

```ts
import type { File, Suite } from 'vitest'
```

**Reply by sheremet-va** [reply]
https://github.com/vitest-dev/vitest/blob/7012f8c12a3d0004e7e7651bbeee905ac9102098/packages/runner/src/types/tasks.ts#L65

**Reply by yuzheng14** [reply]
But I couldn't know what does the fields of it mean, I couldn't find any explanation of the type `File` or `Suite`.

**Reply by sheremet-va** [reply]
I don't think there is any public documentation about it. The only documentation is https://vitest.dev/advanced/runner.html

**Reply by yuzheng14** [reply]
There is type `Suite` in this documentation actually, but it doesn't explain it. So is this a internal type that doesn't plan to use externally?

**Reply by sheremet-va** [reply, marked as answer]
It's part of the public runner API, but it's not documented.
